include/boost/json/detail/charconv/detail/config.hpp

```diff
@@ -0,0 +1,57 @@
+// Copyright 2023 Matt Borland
+// Distributed under the Boost Software License, Version 1.0.
+// https://www.boost.org/LICENSE_1_0.txt
+
+#ifndef BOOST_JSON_DETAIL_CHARCONV_DETAIL_CONFIG_HPP
+#define BOOST_JSON_DETAIL_CHARCONV_DETAIL_CONFIG_HPP
+
+#include <boost/config.hpp>
+#include <type_traits>
+#include <cfloat>
+
+// Use 128 bit integers and supress warnings for using extensions
+#if defined(BOOST_HAS_INT128)
+#  define BOOST_JSON_INT128_MAX  (boost::int128_type)(((boost::uint128_type) 1 << 127) - 1)
+#  define BOOST_JSON_UINT128_MAX ((2 * (boost::uint128_type) BOOST_JSON_INT128_MAX) + 1)
+#endif
+
+#ifndef BOOST_NO_CXX14_CONSTEXPR
+#  define BOOST_JSON_CXX14_CONSTEXPR BOOST_CXX14_CONSTEXPR
+#else
+#  define BOOST_JSON_CXX14_CONSTEXPR inline
+#endif
```

Comment:
```suggestion
#ifndef BOOST_NO_CXX14_CONSTEXPR
#  define BOOST_JSON_CXX14_CONSTEXPR BOOST_CXX14_CONSTEXPR
#  define BOOST_JSON_CXX14_CONSTEXPR_NO_INLINE BOOST_CXX14_CONSTEXPR
#else
#  define BOOST_JSON_CXX14_CONSTEXPR inline
#  define BOOST_JSON_CXX14_CONSTEXPR_NO_INLINE
#endif
```

You'll need the no inline version in fast float because I think everything marked `CXX14_CONSTEXPR` is also `BOOST_FORCEINLINE`. They'll collide with C++11

Reply:
Indeed